src/parser/NormalizedString.h

```diff
@@ -8,9 +8,14 @@
 #include <string>
 #include <string_view>
 
+#include "backports/three_way_comparison.h"
+
 struct NormalizedChar {
   char c_;
-  auto operator<=>(const NormalizedChar&) const = default;
+  QL_DEFINE_CLASS_MEMBERS_AS_TIE(c_)
+
+  QL_DEFINE_EQUALITY_OPERATOR(NormalizedChar)
+  QL_DEFINE_THREEWAY_OPERATOR(NormalizedChar)
```

Comment:
This should be writable as a single line, e.g.
`QL_DEFINE_DEFAULTED_THREEWAY_OPERATOR(NormalizedChar, c_)`

(the threeway macro should include the `==` and `!=`).

This will make it easier (in a future version), to automatically verify that the list of members is exhaustive (or to even generate this automatically).